dependency-convergence/src/main/java/cloudbomcommitcheck/RecentCommitCheck.java

```diff
@@ -0,0 +1,208 @@
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.commons.cli.ParseException;
+import org.eclipse.aether.artifact.Artifact;
+import org.eclipse.aether.artifact.DefaultArtifact;
+
+public class RecentCommitCheck {
+
+  /**
+   * Identifiers for commits with new releases, or updated dependency versions
+   */
+  private static final String updateDependency = "deps: update dependency com.google.cloud:google-cloud-";
+  private static final String newRelease = "chore: release";
+  private static final String snapshot = "SNAPSHOT";
+
+  private final Map<ArtifactData, ClientLibraryStatus> clientLibraries;
+  private final String commitMessage;
+
+  public static void main(String[] args) throws ParseException, MavenRepositoryException {
+    if (args.length < 1) {
+      System.out.println("Please pass a commit message to run this script!");
+      System.exit(1);
+      return;
+    }
+    RecentCommitCheck commitCheck = new RecentCommitCheck(args[0]);
+    int exitCode = commitCheck.execute();
+    commitCheck.printOutput(exitCode);
+    System.exit(exitCode);
+  }
+
+  public RecentCommitCheck(String commitMessage) {
+    this.commitMessage = commitMessage;
+    this.clientLibraries = new HashMap<>();
+  }
+
+  public Map<ArtifactData, ClientLibraryStatus> getCurrentClientLibraries() {
+    return new HashMap<>(clientLibraries);
+  }
+
+  /**
+   * @return program exit code - 0 for success, 1 for invalid dependencies, 2 for unable to find
+   * latest version of google-cloud-shared-dependencies
+   */
+  public int execute() throws ParseException, MavenRepositoryException {
+    if (commitMessage == null || (!commitMessage.contains(updateDependency)
+        && !commitMessage.contains(newRelease)) || commitMessage.contains(snapshot)) {
+      System.out.println("Commit message does not update dependencies. Returning success.");
+      return 0;
+    }
+
+    DefaultArtifact latestSharedDependencies = new DefaultArtifact("com.google.cloud",
+        "google-cloud-shared-dependencies", null, null);
+    ArtifactData sharedDependenciesData = ArtifactData
+        .generateArtifactData(latestSharedDependencies);
+    String latestSharedDependenciesVersion = sharedDependenciesData.getLatestVersion();
+    if (latestSharedDependenciesVersion == null || latestSharedDependenciesVersion.isEmpty()) {
+      return 2;
+    }
+
+    // If we expect to run this twice, clear old dependency data
+    clientLibraries.clear();
+
+    if (commitMessage.contains(updateDependency)) {
+      String dependencyStart = commitMessage.substring(commitMessage.indexOf("com.google.cloud:"));
+      // Should be of the form ["groupId:artifactId", "to", "vX.X.X"]
+      String[] items = dependencyStart.split(" ");
+      if (items.length != 3) {
+        System.out.println("Commit message does not update dependencies. Returning success");
```

Comment:
we're printing the same thing three time below as error msg -- I'm not sure if this will be helpful in debugging. Maybe we could give a bit more information/ be explicit on what the error is.